### ISSUE TITLE
repeat no longer repeats inserts after a deletion

### ISSUE THREAD

**Issue by sunjay** [atom/vim-mode#1006]
One of the awesome things I used to be able to do to quickly replace a word is type `cwFoo` and then use `.` to repeat that everywhere else. Today I restarted atom and installed some updates. This is no longer working now.

### Here's a complete example:

Let's say I have the following lines:

```
var import = {};
import.a = 1;
import.b = 3;
```

If I go to the `i` of the first line, I expect that typing `cwexport<Esc>` will replace `import` with `export`. Going to the beginning of the second and third line and repeating that command with `.`, I expect that each of the other `import`s will be replaced with `export`.

This would look like this:

```
var export = {};
export.a = 1;
export.b = 3;
```

Currently, running those exact commands results in this:

```
var export = {};
.a = 1;
.b = 3;
```

The deletion gets repeated, but the insert doesn't. That would be fine if I was using `d`. With `c`, I expect the insertion to be part of the repeat. 

This works as expected in regular vim.

**Comment by sunjay** [comment]
Any update on this? This is still happening.

**Comment by scotte** [comment]
Same issue here - "." is a pretty fundamental and important vi binding...

**Comment by horiuchi** [comment]
I have same issue too.

Atom 1.7.2
vim-mode 0.65.0
MacOSX

**Comment by jessejanderson** [comment]
Same issue here, as @scotte said this is a pretty fundamental binding...

Atom 1.7.1
vim-mode 0.65.0
Mac OS X El Capitan

**Comment by jessejanderson** [comment]
Co-worker exhibits the same issue with Atom 1.7.0 and vim-mode 0.63.0

**Comment by morenoh149** [comment]
seeing this as well. No recent commits here seem related https://github.com/atom/vim-mode/commits/master could it be an interaction between installed packages?

**Comment by sunjay** [comment]
Doesn't seem like it. Only vim-mode is acting on that key.

![ss 2016-04-21 at 03 36 58](https://cloud.githubusercontent.com/assets/530939/14721904/fbfe1804-07d6-11e6-908d-7c84290bc7c5.png)

**Comment by morenoh149** [comment]
Don't get how the latest commit was on march 10th and this wasn't reported until April 14th though...

**Comment by sunjay** [comment]
Maybe something in the core broke it?

**Comment by morenoh149** [comment]
then maybe 1.7.0 or 1.7.1 broke it

https://github.com/atom/atom/releases/tag/v1.7.0

https://github.com/atom/atom/releases/tag/v1.7.1

**Comment by jessejanderson** [comment]
I think I've been experiencing it for a few weeks now...

**Comment by t9md** [comment]
This is not because of Atom core broken, it's because of Atom core changed to improve(It's always happen).
Underlying text-buffer package was updated on Atom v1.7.0, specifically editor.buffer.history stack changed it's representation of changes.

So `item.confirmChages` no longer works as like before.

https://github.com/atom/vim-mode/blob/master/lib/vim-state.coffee#L456-L460

To fix that, essentially `TransactionBundler` in `operators/input.coffee` must be updated to correctly bundle bunch of changes from checkpoint you start `insert-mode`.

For people who cannot wait someone pick to fix this issue, other option is to try vim-mode-plus(fork of vim-mode) which I'm maintaining actively, this issue is already fixed when Atom v1.7.0 was released.

**Comment by ctbarrett** [comment]
+1

**Comment by haasDev** [comment]
+1

**Comment by b6pzeusbc54tvhw5jgpyw8pwz2x6gs** [comment]
+1

**Comment by jskopek** [comment]
+1. Didn't realize how important this key was until I no longer had access to it.

**Comment by Jakeah** [comment]
+1 bump to high heaven

**Comment by jeffballard** [comment]
+1 for repeating goodness

**Comment by gnebehay** [comment]
+1 same here

**Comment by gjngeldenhuis** [comment]
+1

**Comment by morenoh149** [comment]
Please add to the discussion. Perform a little digging and surface your results. +1's aren't helping anyone.

**Comment by Jakeah** [comment]
@t9md has already stated the issue and what the fix needs to be.  The +1's are hopefully bringing attention to the issue and that it needs to be pushed sooner than later.

**Comment by morenoh149** [comment]
k, so we change [TransactionBundler](https://github.com/atom/vim-mode/blob/a2464a8c434dfbe6db9b2ca99f6a49c50a4d8d31/lib/operators/input.coffee#L140) to *"bundle bunch of changes from checkpoint you start `insert-mode`"*. Which I'm not 100% sure what that means.

**Comment by t9md** [comment]
Here is the commit I fixed this issue in vim-mode-plus.

https://github.com/t9md/atom-vim-mode-plus/commit/76b710260bc795aceee7d3a32c7a02cc18320a3a

But this can't directly be applied to vim-mode.
I removed TransactionBundler far long before since it seems unnecessary for me.

Code base is completely different, but if you really want to fix vim-mode, there is many hint in vim-mode-plus.
I once interested to contribute directly to vim-mode and I did some of contribution. But now I dont want to spend same fix iteration for the issue I've already fixed in vim-mode-plus.

**Comment by psifertex** [comment]
@Jakeah Now that github supports adding reaction emoji to comments and bugs, just add your vote to the original issue with a thumbs up. They are sortable in issue lists so others can see what has interest, but it doesn't spam other people who have subscribed to this thread and want actual updates.

**Comment by BinaryMuse** [comment]
Fixed by #1020. Thanks very much for your patience!
